_rules/explicit-SVG-image-has-name-7d6734.md

```diff
@@ -0,0 +1,180 @@
+
+Browser and assistive technology support for SVG `<title>` and `<desc>` elements is currently inconsistent. Using WAI ARIA in combination with the `img` role for non-decorative `<svg>` elements significantly improves accessibility browser support.
+
+Until browser support for the [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0) is more consistent it is recommended to explicitly remove decorative <svg> elements from the accessibility tree.
+
+## Background
+
+- [Understanding Success Criterion 1.1.1: Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
+- [SVG Accessibility API Mappings (working draft)](https://www.w3.org/TR/svg-aam-1.0)
+- [WAI-ARIA Graphics Module](https://www.w3.org/TR/graphics-aria-1.0/)
+- [ARIA4: Using a WAI-ARIA role to expose the role of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4)
+- [ARIA6: Using aria-label to provide labels for objects](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6)
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `svg` element has an explicit role of `img` and an accessible name from the `title` element that is not empty.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<title>1 circle</title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Passed Example 2
+
+The `circle` element has an explicit role of `graphics-symbol` and an accessible name from the `aria-label` attribute that is not empty.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="https://www.w3.org/2000/svg">
+	<circle
+		role="graphics-symbol"
+		cx="50"
+		cy="50"
+		r="40"
+		stroke="green"
+		stroke-width="4"
+		fill="yellow"
+		aria-label="1 circle"
+	></circle>
+</svg>
+```
+
+#### Passed Example 3
+
+The `svg` element has an explicit role of `graphics-document` and an accessible name from the `title` element that is not empty.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="graphics-document" width="100" height="100">
+	<title>1 circle</title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `svg` element has an explicit role of `img` but has no accessible name.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed Example 2
+
+The `svg` element has an explicit role of `img` but has only whitespace in the `title` element for the accessible name.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img">
+	<title></title>
```

Comment:
Just a note that this example does not have whitespace within the title element as per the description. It is identical code to Failed Example 3.